### PR TITLE
[UX] Simplify the "Allow CORS" action in SPA create wizard

### PR DESCRIPTION
**Pull request by yasinmiran**
## Implementation
https://user-images.githubusercontent.com/25561152/104960380-19886f80-59fa-11eb-8649-364b03edc01d.mp4

### CODE REVIEW

File: modules/core/src/models/url-components.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+export interface URLComponents {
```
Comment (brionmario) [inline modules/core/src/models/url-components.ts:1]: Add WSO2 license header.

File: modules/core/src/utils/url-utils.ts [hunk 2]
```diff
@@ -87,4 +88,36 @@ export class URLUtils {
...
+    };
+
+    public static isTLSEnabled(url: string): boolean {
+        return Boolean(URLUtils.urlComponents(url)
+            .protocol
+            .match("https")
+            ?.length
+        );
+    }
+
```
Comment (brionmario) [inline modules/core/src/utils/url-utils.ts:122]:
```suggestion
```

File: modules/theme/src/themes/default/elements/button.overrides [hunk 3]
```diff
@@ -57,6 +57,13 @@
                 background: @downBackgroundColor;
             }
         }
+        &.with-no-border {
```
Comment (brionmario) [inline modules/theme/src/themes/default/elements/button.overrides:60]: Since we use `LinkButton` do we need this?

**Review by brionmario — APPROVED** [review]

File: modules/core/src/models/url-components.ts (hunk 1 above)
Comment (brionmario) [inline modules/core/src/models/url-components.ts:1]:
```suggestion
export interface URLComponentInterface {
```

Comment (yasinmiran) [reply modules/core/src/models/url-components.ts]: +1 fixed in commit 514413c

File: modules/core/src/utils/url-utils.ts (hunk 2 above)
Comment (yasinmiran) [reply modules/core/src/utils/url-utils.ts]: +1 fixed in commit de77eec

File: modules/core/src/models/url-components.ts (hunk 1 above)
Comment (yasinmiran) [reply modules/core/src/models/url-components.ts]: +1 fixed in commit 294fb24

File: modules/theme/src/themes/default/elements/button.overrides (hunk 3 above)
Comment (yasinmiran) [reply modules/theme/src/themes/default/elements/button.overrides:60]: Yes, we do need it. I have referenced it as an inline className

```suggestion
                            className={ "m-1 p-2 with-no-border orange" }
```

**Pull request merged**